test/test_common/wasm_base.h

```diff
@@ -145,13 +143,9 @@ template <typename Base = testing::Test> class WasmHttpFilterTestBase : public W
     auto wasm = WasmTestBase<Base>::wasm_ ? WasmTestBase<Base>::wasm_->wasm().get() : nullptr;
     int root_context_id = wasm ? wasm->getRootContext(WasmTestBase<Base>::plugin_, false)->id() : 0;
     context_ = std::make_unique<TestFilter>(wasm, root_context_id, WasmTestBase<Base>::plugin_);
-    context_->setDecoderFilterCallbacks(decoder_callbacks_);
-    context_->setEncoderFilterCallbacks(encoder_callbacks_);
```

Comment:
What are we gaining from moving this from `WasmHttpFilterTestBase` to individual tests? Since the code added in individual tests follows `setupFilter()`, I don't see it changes much?

It seems that the only real change here is that we're dropping `NiceMock<>`.